src/TableFunctions/TableFunctionFactory.h

```diff
@@ -61,8 +78,14 @@ class TableFunctionFactory final: private boost::noncopyable, public IFactoryWit
 
     String getFactoryName() const override { return "TableFunctionFactory"; }
 
+    std::optional<TableFunctionProperties> tryGetPropertiesImpl(const String & name) const;
+
     TableFunctions table_functions;
     TableFunctions case_insensitive_table_functions;
+
+    inline static const NameSet table_functions_allowed_in_readonly_mode = {
```

Comment:
I am still not sure whether having NameSet with some property outside of these functions files is a good thing. I originally thought we would make it similar to Aggregate Functions, and for example go from this:
``` cpp
void registerTableFunctionView(TableFunctionFactory & factory)
{
    factory.registerFunction<TableFunctionView>();
}
```
to this:
``` cpp
void registerTableFunctionView(TableFunctionFactory & factory)
{
    factory.registerFunction<TableFunctionView>({{}, .allow_readonly = true});
}
```

Reply:
Originally I didn't want to duplicate the logic of checking the NameSet in all registerFunction calls, so I added this in factory method, but on second thought, I agree with your suggestion that its not a good idea to keep this logic in factory method.
`template <typename Function>
    void registerFunction(Documentation doc = {}, bool allow_readonly = false, CaseSensitiveness case_sensitiveness = CaseSensitive)`

I added the NameSet so that its convenient for us to add/remove readonly functions in future (instead of checking all registerFunction calls). I can add a function 
`static bool isReadOnly(String name) const `

and we can use that 

`void registerTableFunctionView(TableFunctionFactory & factory)
{
    factory.registerFunction<TableFunctionView>({{}, isReadOnly(name)});
}`

Please let me know your thoughts on this.

Reply:
It seems like usually in our codebase parameters for different classes (Aggregate Functions, QueryPlan Steps) are set inside files of those classes. For me personally, it would be strange to have a NameSet with the names of different subclasses (which are set manually and do not automatically come from TableFunction***::getName()) to derive their parameters.

Reply:
> I added the NameSet so that its convenient for us to add/remove readonly functions in future (instead of checking all registerFunction calls).

It also seems like the change is quite simple and we only need to do it when we assign true to the `allow_readonly` parameter, otherwise, we will call the default TableFunctionProperties constructor and this parameter will be automatically assigned false

> For example, go from this:
> ``` cpp
> void registerTableFunctionView(TableFunctionFactory & factory)
> {
>     factory.registerFunction<TableFunctionView>();
> }
> ```
> to this:
> ``` cpp
> void registerTableFunctionView(TableFunctionFactory & factory)
> {
>     factory.registerFunction<TableFunctionView>({{}, .allow_readonly = true});
> }
> ```

Reply:
cool, I agree its better be good to keep similar pattern. will update this

Reply:
addressed it